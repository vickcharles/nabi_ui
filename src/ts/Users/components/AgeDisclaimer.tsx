import * as React from 'react';

import Button from 'material-ui/Button';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';

interface AgeDisclaimerProps {
  isFormDialogOpen: boolean;
  closeHandler: () => void;
}

const AgeDisclaimer: React.StatelessComponent <AgeDisclaimerProps> = props => {
  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">AGE DISCLAIMER NOTICE</DialogTitle>
        <DialogContent>
            <DialogContentText>
              You must be 13 years of age or older to create an account. Please have
              your parent or guardian assist you. For more assistance contact us at
              info@nabimusic.com.
            </DialogContentText>
        </DialogContent>
        <DialogActions>
        <Button variant="raised" onClick={props.closeHandler} color="primary">
          Okay
        </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default AgeDisclaimer;
