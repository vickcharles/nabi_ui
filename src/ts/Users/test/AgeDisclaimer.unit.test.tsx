import * as React from 'react';
import AgeDisclaimer from '../components/AgeDisclaimer';
import { shallow } from 'enzyme';

describe('AgeDisclaimer', () => {
  let wrapper: any;
  const closeHandler: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <AgeDisclaimer
          isFormDialogOpen={false}
          closeHandler={closeHandler}
        />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
