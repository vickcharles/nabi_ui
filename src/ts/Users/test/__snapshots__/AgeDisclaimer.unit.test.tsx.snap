// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeDisclaimer Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AgeDisclaimer
    closeHandler={[MockFunction]}
    isFormDialogOpen={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="form-dialog-title"
        onClose={[MockFunction]}
        open={false}
      >
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          AGE DISCLAIMER NOTICE
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            onClick={[MockFunction]}
            variant="raised"
          >
            Okay
          </WithStyles(Button)>
        </WithStyles(DialogActions)>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            AGE DISCLAIMER NOTICE
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="raised"
            >
              Okay
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "AGE DISCLAIMER NOTICE",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "AGE DISCLAIMER NOTICE",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)>
              You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.",
            },
            "ref": null,
            "rendered": "You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="raised"
            >
              Okay
            </WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Okay",
              "color": "primary",
              "onClick": [MockFunction],
              "variant": "raised",
            },
            "ref": null,
            "rendered": "Okay",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="form-dialog-title"
          onClose={[MockFunction]}
          open={false}
        >
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            AGE DISCLAIMER NOTICE
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="raised"
            >
              Okay
            </WithStyles(Button)>
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "form-dialog-title",
          "children": Array [
            <WithStyles(DialogTitle)
              id="form-dialog-title"
            >
              AGE DISCLAIMER NOTICE
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)>
                You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
              </WithStyles(DialogContentText)>
            </WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="raised"
              >
                Okay
              </WithStyles(Button)>
            </WithStyles(DialogActions)>,
          ],
          "onClose": [MockFunction],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "AGE DISCLAIMER NOTICE",
              "id": "form-dialog-title",
            },
            "ref": null,
            "rendered": "AGE DISCLAIMER NOTICE",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)>
                You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.
              </WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.",
              },
              "ref": null,
              "rendered": "You must be 13 years of age or older to create an account. Please have your parent or guardian assist you. For more assistance contact us at info@nabimusic.com.",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="raised"
              >
                Okay
              </WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Okay",
                "color": "primary",
                "onClick": [MockFunction],
                "variant": "raised",
              },
              "ref": null,
              "rendered": "Okay",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
